src/test/java/xyz/groundx/caver_ext_kas/kas/wallet/WalletAPITest.java

```diff
@@ -2739,4 +2745,308 @@ public void checkResponse() throws ExecutionException, InterruptedException {
+
+        assertNotNull(keySignDataResponse);
+    }
+
+    @Ignore
+    public void signMessageAsync() throws ApiException, ExecutionException, InterruptedException {
+        String data = "0x1122334455667788112233445566778811223344556677881122334455667788";
+        KeyCreationResponse response = caver.kas.wallet.createKeys(1);
+
+        CompletableFuture<KeySignDataResponse> future = new CompletableFuture<>();
+        Call call = caver.kas.wallet.signMessageAsync(response.getItems().get(0).getKeyId(), data, new ApiCallback<KeySignDataResponse>() {
+            @Override
+            public void onFailure(ApiException e, int statusCode, Map<String, List<String>> responseHeaders) {
+                future.completeExceptionally(e);
+            }
+
+            @Override
+            public void onSuccess(KeySignDataResponse result, int statusCode, Map<String, List<String>> responseHeaders) {
+                future.complete(result);
+            }
+
+            @Override
+            public void onUploadProgress(long bytesWritten, long contentLength, boolean done) {
+
+            }
+
+            @Override
+            public void onDownloadProgress(long bytesRead, long contentLength, boolean done) {
+
+            }
+        });
+
+        if(future.isCompletedExceptionally()) {
+            fail();
+        } else {
+            assertNotNull(future.get());
+        }
+    }
+
+    private FeeDelegatedAccountUpdate createAccountUpdateTx(SingleKeyring keyring, String newKey) throws IOException {
+        com.klaytn.caver.account.Account account = com.klaytn.caver.account.Account.createWithAccountKeyPublic(keyring.getAddress(), newKey);
+
+        FeeDelegatedAccountUpdate tx = new FeeDelegatedAccountUpdate.Builder()
+                .setKlaytnCall(caver.rpc.klay)
+                .setFrom(keyring.getAddress())
+                .setGas(BigInteger.valueOf(250000))
+                .setAccount(account)
+                .build();
+
+        tx.sign(keyring);
+
+        return tx;
+    }
+
+    @Ignore
+    public void registerAccounts() throws ApiException, IOException {
+        KeyCreationResponse response = caver.kas.wallet.createKeys(1);
+
+        SingleKeyring keyring = KeyringFactory.generate();
+        com.klaytn.caver.account.Account account = com.klaytn.caver.account.Account.createWithAccountKeyLegacy(keyring.getAddress());
+
+        AccountRegistration registration = new AccountRegistration();
+        registration.setAddress(account.getAddress());
+        registration.setKeyId(response.getItems().get(0).getKeyId());
+        registration.setRlp(createAccountUpdateTx(keyring, response.getItems().get(0).getPublicKey()).getRLPEncoding());
+
+        AccountRegistrationRequest request = new AccountRegistrationRequest();
+        request.add(registration);
+
+        RegistrationStatusResponse result = caver.kas.wallet.registerAccounts(request);
```

Comment:
How about checking status in result?
If registration is fail, KAS returns failure in response object.

Reply:
Okay, I added logic that check registration status.